RSyntaxTextArea/src/main/java/org/fife/ui/rtextarea/LineNumberList.java

```diff
@@ -726,7 +739,10 @@ void updateCellWidths() {
 					lineCount = lineCount/10;
 					count++;
 				} while (lineCount >= 10);
-				cellWidth += fontMetrics.charWidth('9')*(count+1) + 3;
+				// This was required since the current line number has a higher font size so the previous method can cause the first digit of the
+				// number to go out of bounds of the line number list.
+				// Maybe there is some better solution to this situation.
+				cellWidth += fontMetrics.charWidth('9')*(count+1) + (fontMetrics.charWidth('9') * (int)currentLineNumberFontIncreaseSize);
```

Comment:
Assuming we just move to a `currentLineNumberFont` variable, we can just pick the larger of the two `FontMetrics` values here to simplify things a bit:

```java
Font font = getFont();
FontMetrics fontMetrics = font.getFontMetrics();
int charWidth = fontMetrics.charWidth('9');
if (currentLineNumberFont != null) {
  int charWidth2 = currentLineNumberFont.getFontMetrics().charWidth('9');
  charWidth = Math.max(charWidth, charWidth2);
}
...
cellWidth += charWidth * (count + 1) + 3;
```

Might even move that `charWidth` calculation logic into a private helper method just to make the code more readable,.